src/queueserver.ts

```diff
@@ -117,3 +146,43 @@ export const clearQueue = async(): Promise<IPlan> => {
     console.log(res);
     return res.data;
 }
+
+export interface IPlanModifyState {
+    readonly modify: IPlanModify;
+    readonly loading: boolean;
+}
+
+export const modifyEnvironment = async(op: number): Promise<IPlanModify> => {
+    var operation = "open";
+    if (op === 0) {
+        operation = "open";
+    }
+    else if (op === 1) {
+        operation = "close";
+    }
+    else if (op === 2) {
+        operation = "destroy";
+    }
```

Comment:
Would it be more reliable to have an enum with the mappings?

Reply:
Probably, I was being lazy and getting something to work. I will take another look, I was happy to see it working. This needs better state management too, it is pretty fragile right now!

Reply:
I made the change, the way TypeScript implements enums is quite nice, so I used the string enums with lookup to simplify the code quite a bit too.

Reply:
Thank you.